src/Mono.Android/Android.Runtime/JNIEnv.cs

```diff
@@ -239,6 +239,11 @@ internal static unsafe void Initialize (JnienvInitializeArgs* args)
 					Java.Lang.Thread.DefaultUncaughtExceptionHandler = defaultUncaughtExceptionHandler;
 			}
 
+			var packageNamingPolicy = Environment.GetEnvironmentVariable ("__XA_PACKAGE_NAMING_POLICY__");
+			if (Enum.TryParse (packageNamingPolicy, out PackageNamingPolicy pnp)) {
+				JavaNativeTypeManager.PackageNamingPolicy = pnp;
+			}
```

Comment:
@garuma if the designer can get the value of the `$(AndroidPackageNamingPolicy)` MSBuild property and pass it as an environment variable named `__XA_PACKAGE_NAMING_POLICY__`, I think that will get things working properly.

Reply:
Thanks @garuma we're green!